bundles/org.openhab.binding.bluetooth.generic/src/main/java/org/openhab/binding/bluetooth/generic/internal/GenericBluetoothHandler.java

```diff
@@ -0,0 +1,431 @@
+                if (fields.size() == 1) {
+                    label = gattChar.getName();
+                }
+
+                Map<String, List<Field>> fieldsMapping = fields.stream().collect(Collectors.groupingBy(Field::getName));
+
+                for (List<Field> fieldList : fieldsMapping.values()) {
+                    Field field = fieldList.get(0);
+                    if (fieldList.size() > 1) {
+                        if (field.isFlagField() || field.isOpCodesField()) {
+                            logger.debug("Skipping flags/op codes field: {}.", charUUID);
+                        } else {
+                            logger.warn("Multiple fields with the same name found: {} / {}. Skipping these fields.",
+                                    charUUID, field.getName());
+                        }
+                        continue;
+                    }
+
+                    if (isFieldSupported(field)) {
+                        Channel channel = buildFieldChannel(field, label, !gattChar.isValidForWrite());
+                        if (channel != null) {
+                            channels.add(channel);
+                        } else {
+                            logger.warn("Unable to build channel for field: {}", field.getName());
+                        }
+                    } else {
+                        logger.warn("GATT field is not supported: {} / {} / {}", charUUID, field.getName(),
+                                field.getFormat());
+                    }
+                }
+            } else {
+                channels.add(buildUnknownChannel());
+            }
+            return channels;
+        }
+
+        private Channel buildUnknownChannel() {
+            ChannelUID channelUID = getChannelUID(null);
+            ChannelTypeUID channelTypeUID = new ChannelTypeUID(BluetoothBindingConstants.BINDING_ID, "char-unknown");
+            return ChannelBuilder.create(channelUID).withType(channelTypeUID).withProperties(getChannelProperties(null))
+                    .build();
+        }
+
+        public boolean canRead() {
+            String charUUID = getCharacteristicUUID();
+            if (gattParser.isKnownCharacteristic(charUUID)) {
+                return gattParser.isValidForRead(charUUID);
+            }
+            // TODO: need to evaluate this from characteristic properties, but such properties aren't support yet
+            return true;
+        }
+
+        public boolean canWrite() {
+            String charUUID = getCharacteristicUUID();
+            if (gattParser.isKnownCharacteristic(charUUID)) {
+                return gattParser.isValidForWrite(charUUID);
+            }
+            // TODO: need to evaluate this from characteristic properties, but such properties aren't support yet
+            return true;
+        }
+
+        private boolean isAdvanced() {
+            return false;// TODO need to decide whether to implement this or not
```

Comment:
Even I don't know what to do about this since the TODO is from @vkolotov's code that I copied over. It is unknown at this time if/how to implement this.

Reply:
I would suggest to remove it then.

Reply:
I just implemented it instead.